File: Assignment 01/Program 20/prog20.py
#Write a Python program to find whether a given array of integers contains any
#duplicate element. Return true if any value appears at least twice in the said

import array as arr

a1 = arr.array('i')
n = int(input("Enter the no. of elements: "))
for i in range(n):
    val = int(input("Enter the element: "))
    a1.append(val)

unique_set = set()
f = 0

for ele in a1:
    if ele in unique_set:
        f=1
        break
    else:
        unique_set.add(ele)
    

if f==1:
    print("True")
else:
    print("False")

File: Data Structure and Algorithm/Assignment 01/Program 20/prog20.c
//Write a Python program to find whether a given array of integers contains any
//duplicate element. Return true if any value appears at least twice in the said

#include<stdio.h>
#include<stdlib.h>

int main(){
    int n;
    printf("Enter the number of inputs: ");
    scanf("%d",&n);
    int *arr = (int*)malloc(n*sizeof(int));
    for (int i = 0; i < n; i++)
    {
        printf("Enter the number %d: ",i+1);
        scanf("%d",(arr+i));
    }
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        for (int j = i+1; j < n; j++)
        {
            if(arr[i] == arr[j]){
                count+=1;
                break;
            }
            else
                continue;
        }
        
    }
    if(count>0){
        printf("True");
    }
    else    
        printf("False");
        free(arr);
    return 0;
}